Userland/Libraries/LibArchive/TarStream.cpp

```diff
@@ -27,6 +27,8 @@ ErrorOr<Bytes> TarFileStream::read(Bytes bytes)
     auto to_read = min(bytes.size(), header_size - m_tar_stream.m_file_offset);
 
     auto slice = TRY(m_tar_stream.m_stream->read(bytes.trim(to_read)));
+    if (slice.size() != to_read)
+        return Error::from_string_literal("Attepted to read past the end of the stream");
```

Comment:
We have `read_entire_buffer` now, which returns an error if the buffer can't be filled entirely and is also `ErrorOr<>` compatible. :^)

Reply:
That said, we may just want to replace the `VERIFY` in `for_each_extended_header` instead.